docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+````
+_________________________________
+Welcome to scheduler--;!
+What can I do for you?
+_________________________________
+File Created: Personal
+File Created: Zoom
+File Created: Timetable
+File Created: Goal
+The file has successfully been loaded!
+_________________________________
+You have no events today!
+_________________________________
+````
+## Design
+### Architecture
 
+![Architecture Diagram](./diagrams/Architecture.jpg)
 
-## Product scope
-### Target user profile
+The ***Architecture Diagram*** given above explains the high-level design of App. Given below is a quick overview of
+each component.
 
-{Describe the target user profile}
+`Duke` class contains the main method. It is responsible for,
 
-### Value proposition
+- At app launch: Initializes the components in the correct sequence, and connects them up with each other.  
+- At shut down: Stores the user data from memory into the hard disk.
 
-{Describe the value proposition: what problem does it solve?}
+The rest of the App consists of five components.
 
-## User Stories
+- `Ui`: The Ui of the App.
+- `Parser`: Parses the input to determine which command to execute.
+- `Command`: The command executer.
+- `UserData`: Holds the data of the App in the memory.
+- `Storage`: Reads data from, and writes data to, the hard disk.
+
+#### Ui Component
+The 'Ui' component is in charge of handling input from users and system output.
+
+#### Command Component
+
+![Diagram for commmand](./diagrams/Command.jpg)
+
+API: [Command.java](https://github.com/AY2021S1-CS2113T-T12-4/tp/blob/master/src/main/java/seedu/duke/command/Command.java) 
+1. `Parser` class to parse the user command 
+1.  This results in a `Command` object executed by the `Parser`
+1.  The command execution can affect the `Model`
+
+It listens for commands made in the Duke Class and sends the input to the parser class.
+It is also responsible for printing messages from commands and exception messages. 
+
+#### Parser Component
+![Parser diagram](./diagrams/Parser_diagram.jpg)
+
+The `Parser` object uses its `parse` method to parse inputs passed to it from `Ui`. This method returns a `Command` object to be executed by `Duke`.
+
+#### UserData Component
+![UserData diagram](./diagrams/UserData_diagram.jpg)
+
+The `UserData`
+- stores `EventList` objects that store events of different categories (i.e. Personal, Timetable, Zoom).
+- stores a `Goal` object that represents the current user goal.
+- does not depend on the other components.
+
+#### Storage Component
+![Diagram for storage class](./diagrams/classDiagramStorage.jpg)
+
+API [Storage.java](https://github.com/AY2021S1-CS2113T-T12-4/tp/blob/master/src/main/java/seedu/duke/storage/Storage.java) 
+
+The storage component,
+- is able to store all event and goal information onto the computer
+- is able to load all event and goal information from the computer into the program
+
+![Diagram for storageOverall](./diagrams/storageOverall.jpg)
```

Comment:
Consider using sequence diagrams instead of non-standard diagram. More chance to showcase diagramming skills is always good. See below: some other type used ... UML ... worked just as well.
![image](https://user-images.githubusercontent.com/62279011/97544627-d7f0b580-1a04-11eb-8820-25dac4a93fbf.png)